docs/new-architecture-playbook.md

```diff
@@ -0,0 +1,2182 @@
+
+  MyComponentsRegistry(ComponentFactory *delegate);
+
+  private:
+  friend HybridBase;
+
+  static std::shared_ptr<ComponentDescriptorProviderRegistry const>
+  sharedProviderRegistry();
+
+  const ComponentFactory *delegate_;
+
+  static jni::local_ref<jhybriddata> initHybrid(
+      jni::alias_ref<jclass>,
+      ComponentFactory *delegate);
+};
+
+} // namespace react
+} // namespace facebook
+```
+
+2. **Create an implementation file: `MyComponentsRegistry.cpp`**
+
+The file should be placed inside the `src/main/jni` folder alongside `MyComponentsRegistry.h
+
+```cpp
+#include "MyComponentsRegistry.h"
+
+#include <CoreComponentsRegistry.h>
+#include <fbjni/fbjni.h>
+#include <react/renderer/componentregistry/ComponentDescriptorProviderRegistry.h>
+#include <react/renderer/components/rncore/ComponentDescriptors.h>
+#include <react/renderer/components/samplelibrary/ComponentDescriptors.h>
+
+namespace facebook {
+namespace react {
+
+MyComponentsRegistry::MyComponentsRegistry(
+    ComponentFactory *delegate)
+    : delegate_(delegate) {}
+
+std::shared_ptr<ComponentDescriptorProviderRegistry const>
+MyComponentsRegistry::sharedProviderRegistry() {
+  auto providerRegistry = CoreComponentsRegistry::sharedProviderRegistry();
+
+  providerRegistry->add(concreteComponentDescriptorProvider<
+                        RNTMyNativeViewComponentDescriptor>());
+
+  return providerRegistry;
+}
+
+jni::local_ref<MyComponentsRegistry::jhybriddata>
+MyComponentsRegistry::initHybrid(
+    jni::alias_ref<jclass>,
+    ComponentFactory *delegate) {
+  auto instance = makeCxxInstance(delegate);
+
+  auto buildRegistryFunction =
+      [](EventDispatcher::Weak const &eventDispatcher,
+          ContextContainer::Shared const &contextContainer)
+      -> ComponentDescriptorRegistry::Shared {
+    auto registry = MyComponentsRegistry::sharedProviderRegistry()
+                        ->createComponentDescriptorRegistry(
+                            {eventDispatcher, contextContainer});
+
+    auto mutableRegistry =
+        std::const_pointer_cast<ComponentDescriptorRegistry>(registry);
+
+    mutableRegistry->setFallbackComponentDescriptor(
+        std::make_shared<UnimplementedNativeViewComponentDescriptor>(
+            ComponentDescriptorParameters{
+                eventDispatcher, contextContainer, nullptr}));
+
+    return registry;
+  };
+
+  delegate->buildRegistryFunction = buildRegistryFunction;
+  return instance;
+}
+
+void MyComponentsRegistry::registerNatives() {
+  registerHybrid({
+      makeNativeMethod("initHybrid", MyComponentsRegistry::initHybrid),
+  });
+}
+
+} // namespace react
+} // namespace facebook
+```
+
+4. **Load your file in the OnLoad.cpp**
+
+If you followed the TurboModule instructions, you should have a `OnLoad.cpp` file inside the `src/main/jni` folder. There you should add a line to load the `MyComponentsRegistry` class:
+
+```cpp
+#include <fbjni/fbjni.h>
+#include "MyApplicationTurboModuleManagerDelegate.h"
```

Comment:
Also need to add `#include "MyComponentsRegistry.h"`

Reply:
Yup nice catch 👍